src/directConnectManager.ts

```diff
@@ -59,10 +77,47 @@ export class DirectConnectionManager {
           const enabled = configs.get(ENABLE_DIRECT_CONNECTIONS, false);
           logger.debug(`"${ENABLE_DIRECT_CONNECTIONS}" config changed`, { enabled });
           setContextValue(ContextValues.directConnectionsEnabled, enabled);
+          // "Other" container item will be toggled
+          getResourceViewProvider().refresh();
+          // if the Topics/Schemas views are focused on a direct connection based resource, wipe them
+          if (!enabled) {
+            const topicsView = getTopicViewProvider();
+            if (topicsView.kafkaCluster && isDirect(topicsView.kafkaCluster)) {
+              topicsView.reset();
+            }
+            const schemasView = getSchemasViewProvider();
+            if (schemasView.schemaRegistry && isDirect(schemasView.schemaRegistry)) {
+              schemasView.reset();
+            }
+          }
+        }
+      },
+    );
+
+    const connectionsListener: Disposable = getExtensionContext().secrets.onDidChange(
+      async ({ key }: SecretStorageChangeEvent) => {
+        // watch for any cross-workspace direct connection additions/removals
+        if (key === SecretStorageKeys.DIRECT_CONNECTIONS) {
+          const connections = await getResourceManager().getDirectConnections();
+          // if the Topics/Schemas views were focused on a resource whose direct connection was removed,
+          // reset the view(s) to prevent orphaned resources from being used for requests
+          const topicsView = getTopicViewProvider();
+          if (topicsView.kafkaCluster && isDirect(topicsView.kafkaCluster)) {
+            if (!connections.has(topicsView.kafkaCluster.connectionId)) {
+              topicsView.reset();
+            }
+          }
+          const schemasView = getSchemasViewProvider();
+          if (schemasView.schemaRegistry && isDirect(schemasView.schemaRegistry)) {
+            if (!connections.has(schemasView.schemaRegistry.connectionId)) {
+              schemasView.reset();
```

Comment:
I'm on the fence about delegating these to the view providers themselves

Reply:
6 one way ... Neither is perfect. The view controllers will start to stink if more and more connection-type-centric things in there.